torch/nn/modules/loss.py

```diff
@@ -123,7 +123,7 @@ class NLLLoss(_WeightedLoss):
 
     .. math::
         \ell(x, y) = L = \{l_1,\dots,l_N\}^\top, \quad
-        l_n = - w_{y_n} x_{n,y_n}, \quad
+        l_n = - w_{y_n} x_n y_n, \quad
```

Comment:
The `y`s (target) are indices in this class. So the current doc is correct no?

Reply:
`y_n` as an index for the weights seems reasonable but why should the input `x` have `y_n` as an index additional to `n`?

Reply:
`n` is the index of the sample and `y_n` is reading the value of X at the ground truth class.
You can check the doc, the size of `x` is `(N, C)` (with N the batch size and C the number of classes).

Reply:
Ok thank you for the clarification, I misunderstood that - sorry.

Reply:
No worries!